### DISCUSSION TITLE
Caching & thumbnails...

### DISCUSSION THREAD

**Discussion by Zelfapp** [jhomlala/betterplayer#442] [Q&A]
With BetterPlayer, if I have a list of videos using, for example, BetterPlayerListViewPlayer. If I'm trying to be bandwidth (cost) conscientious, does BetterPlayer download every video in the list? Is it lazy loading as videos come into view like a ListBuilder.builder?

Or, do I need to figure out a way to create thumbnails using a frame in my videos that simply displays in a list or gridview to keep bandwidth costs down? Are thumbnails for videos easy to make with BetterPlayer, I'm not seeing that option. Currently, if I want to display video that does not play, I'm still using BetterPlayer button disabling controls to get a "thumbnail", but isn't that downloading the entire video just to display a thumbnail?

I see that we can enable caching, so that is good, I guess there would only be one download if caching is running?

**Comment by jhomlala** [comment]
BetterPlayer downloads video only when it's requested with setupDataSource method in BetterPlayerController. So usage of bandwidth depends on how you're placing your widgets. If you're working with list view builder then it should be lazy loading.

Better Player doesn't support creating thumbnails but you can do it easily with other packages for example: https://pub.dev/packages/video_thumbnail

If cache is enabled then video is placed in cache directory. But you need to know that cache is working as queue. If there's not enough space for another videos then last video will be removed. You can play with cache size on Android.
